apps/src/templates/teacherHomepage/Courses.jsx

```diff
@@ -0,0 +1,78 @@
+import $ from 'jquery';
+import React from 'react';
+import RecentCoursesCollapsible from './RecentCoursesCollapsible';
+import CollapsibleSection from './CollapsibleSection';
+import GradientNavCard from './GradientNavCard';
+import shapes from './shapes';
+import i18n from "@cdo/locale";
+
+const Courses = React.createClass({
+  propTypes: {
+    courses: shapes.courses,
+    englishTeacher: React.PropTypes.bool.isRequired
+  },
+
+  componentDidMount() {
+    if (this.props.englishTeacher) {
+      $('.courseexplorer').appendTo(this.refs.courseExplorer).show();
```

Comment:
Things can behave in strange and unexpected ways when we have both jquery and React battling over what belongs in the dom. As things are written right now, if this component needs to rerender it will probably either (a) throw an error or (b) blow away the contents of courseExplorer and others.

What you want here I think is `ProtectedStatefulDiv` https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/ProtectedStatefulDiv.jsx. What this does is it allows React to control the initial render, but states that after that it will make no rerenders. If you were to make your courseExplorer a protected div, you can use jquery to manipulate its contents (as you do here) without having to worry about them being blown away by a future rerender.